### PR TITLE
OGM-1411 Test pagination

### PR DESCRIPTION
**Pull request by DavideD**
Folllows https://github.com/hibernate/hibernate-ogm/pull/977

https://hibernate.atlassian.net/browse/OGM-1411

@fax4ever can you look at this and if it works for you, merge it?

I started to rename the methods but I realize it would be nice to move the tests to core so that we can run the same tests for Neo4j and for JPQL.  I also refactored the code  a bit.

Thanks

### CODE REVIEW

**Comment by fax4ever** [comment]
I like the idea to move the tests to core.
Thank you @DavideD. I'm going to check it.

File: core/src/test/java/org/hibernate/ogm/backendtck/queries/pagination/PaginationUseCases.java [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+import org.hibernate.Session;
+import org.hibernate.ogm.utils.OgmTestCase;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+/**
+ * Test the pagination for native queries with MongoDB
```
Comment (fax4ever) [inline core/src/test/java/org/hibernate/ogm/backendtck/queries/pagination/PaginationUseCases.java]: this test is now generic, maybe we could remove the reference to the dialect

Comment (DavideD) [reply core/src/test/java/org/hibernate/ogm/backendtck/queries/pagination/PaginationUseCases.java]: Updated and pushed force

**Comment by fax4ever** [comment]
I merge it. Thanks @DavideD

**Comment by DavideD** [comment]
@fax4ever Thanks, did you merge it in the 5.3 branch as well?

**Comment by fax4ever** [comment]
Of course. I'm going to backporting it to 5.3 branch.

**Pull request merged**